My submission js2

Ousman37 opened the pull request:
Please can you review My social media app auth? 



Thanks:)

Commit:
installation

package.json,
sass  express eslint nodemon

Commit:
Adding markup

index files created

Commit:
sass compille

Scss files created

Commit:
Implementing

Just workflow  js file created

Fermain commented:
There are 5000 changes, so I am unable to review at this stage. Please remove node_modules from your history and gitignore this at all times.

Commit:
added

.gitignore file
eslintrc.json

Fermain commented on .gitignore:
```diff
@@ -0,0 +1 @@
+/node_modules
```
You need to delete node_modules from history before adding this, comment it out, then delete the folder, commit that, then add this back again.

Ousman37 replied:
OK, let go and delete it then.

Commit:
node_modules

has been removed

Commit:
Added

node_modules  and gitignore file

Ousman37 commented:
I have deleted it and added a new one again.

 Please let me know if it is fine now.

Fermain commented:
Looks good now, will review shortly

Ousman37 commented:
Thank you:)  I had it in my previous set-up. Everything was fine, but then I have trouble with my commit in the terminal.  I try the way you showed us working on three different branches, and it did not work for me. I had some conflicts with the branches. 

After this, I will practice more with small branches to make a commit different branches

Commit:
Media Queries

Working on pages responsive

Commit:
SCSS

working on  media queries

Commit:
Updating markup

working on not found pages

Commit:
Responsive

implementing

Commit:
Media query

implementing

Ousman37 committed:
Update README.md

markdown documentation

Ousman37 committed:
Add files via upload

Added screenshot image

Ousman37 committed:
Update README.md

Ousman37 committed:
Update README.md

Added image

Ousman37 committed:
Update README.md

Ousman37 committed:
Update README.md

Ousman37 committed:
Update README.md

Ousman37 committed:
Update README.md

Ousman37 committed:
Update README.md

Ousman37 committed:
Update README.md

Ousman37 committed:
Update README.md

Ousman37 commented:
Hello man, 

Hope you did not forget me here. 

Have a great weekend out there!

Fermain commented on server.js:
```diff
@@ -0,0 +1,41 @@
+/* eslint-disable no-undef */
+const express = require('express');
```
Are you using express instead of live server?

Ousman37 replied:
Yes, I am using express is okay?

Fermain replied:
Yeah its ok

Commit:
Implementing scss

change the header background colour

Commit:
Merge branch 'workflow_Js2' of https://github.com/Ousman37/Social_medi_-app_Auth-CA into workflow_Js2

Fermain commented on src/create-post.html:
```diff
@@ -0,0 +1,121 @@
+            <input id="media" name="Media" required type="text" class="form-control"
+              placeholder="Enter a image link for Banner" />
+          </div>
+          <button id="create-submit" type="button" class="btn btn-primary">Create Entry</button>
```
```suggestion
          <button id="create-submit" class="btn btn-primary">Create Entry</button>
```

Type button doesn't trigger a submit, have you tested this?

Ousman37 replied:
Yes I have tested it and it works with the type it doesn´t trigger I have Tried your suggestion, but then I couldn't log in.

Fermain commented on src/edit.html:
```diff
@@ -0,0 +1,113 @@
+              placeholder="Enter a image link for Banner" />
+
+          </div>
+          <button id="edit-submit" type="button" class="btn btn-primary">Save</button>
```
```suggestion
          <button id="edit-submit" class="btn btn-primary">Save</button>
```

Fermain commented on src/login.html:
```diff
@@ -0,0 +1,98 @@
+                <li class="uppercase">Must contain an uppercase letter.</li>
+                <li class="special">Must contain a number or special character.</li>
+              </ul>
+              <button type="button" id="login-submit"
```
```suggestion
              <button id="login-submit"
```

Fermain commented on src/register.html:
```diff
@@ -0,0 +1,129 @@
+            <p class="alert alert-danger"> Must be at least 8 characters</p>
+          </div>
+
+          <button type="button" id="register-submit" class="btn_fourth">Register</button>
```
```suggestion
          <button id="register-submit" class="btn_fourth">Register</button>
```

Fermain commented:
I can't see a way to create a post in the UI

Ousman37 committed:
Update src/login.html

I have made the change you suggested, but I am not able to log in without the type="button"

Co-authored-by: Oliver Dipple <Fermain@users.noreply.github.com>

Ousman37 commented:
I think I know why you couldn't see create a post in the UI I am going to fix that.

Commit:
Implementing Scss

  added mobile menu

Commit:
Issue with the button

added the type button back I am not able to login with the type button!

Ousman37 commented:
I have fixed the issue with the create a post in the UI. You will able to see it now. 

let me know.

Thanks :)

Commit:
implementing  scss

Updating changes

Commit:
Implementing

Remove the h1 class

Commit:
implementing Form

Working on the form validation

Commit:
Implementing

 login page  helper text changed to yellow colour

Commit:
Implementing

Working on media queries

Commit:
Implementing sass nesting

media queries!

Commit:
Implementing

Updating changes

Commit:
Implementing

Updating changes

Ousman37 commented:
Hello man,  
I fixed the issue with the Buttons now all submit buttons are formatted correctly.  I guess I found what I was doing wrong in the  HTML form validation have tested it it works properly. 
 
Please let me know if there are any future changes  I could do will be glad to do them. 
Thanks :) 
Ethmane

Fermain commented:
I'm testing now:

- [x] Delete Post (works but not clear to the user)
- [x] Update Post?
- [x] Profile?

Ousman37 commented:
Missing that part. Thank you for pointing that out. 
I’m going to work on that now. luckily, I still have  some times

Commit:
Implementing

Have update all changes :)

Ousman37 commented:
Hello man, 

 I have updated the update post page the profile page is now updated. 
The deleted post  now gives the user a "message" that the post  has been  detected 
I have tested things are working fine. 

 Please let me know in case I'm blind by something. 

Thank you :)

Commit:
Implementing

Just added Follow and unfollow.

Commit:
Implementing

working on js buttons

Commit:
Implementing

I remove some  style on the View and Update  button

Fermain commented:
OK, I think you are just down to UX improvements now. Required functionality is in place but life as a user is not very easy. To improve this:

- Add headings to pages to make it clear what each page is for
- Make sure your link addresses are 100%, I got into a pickle by clicking through the links
- Redirect users after a successful delete or update

Ousman37 commented:
Thank you, man 😎

 I am going to fix that.

Commit:
UX improvements

Fixing

Commit:
Implementing

Improving the UI / UX

Commit:
Implementing

Improving the UI / UX

Commit:
Implementing

Improving the UI / UX

Commit:
Implementing

Improving the UI / UX

Commit:
Implementing

Improving the UI / UX

Commit:
Implementing

Commit:
Implementing

UX/UI

Commit:
Implementing

UI/UX

Commit:
Implementing

Improving the  UI / UX

Commit:
Implementing

Improving UI / UX

Ousman37 commented:
I have improved the UI / UX  now there’s a clear message appealing on the page for the user when they delete the post.

Commit:
Implementing

Improving UI / UX

Commit:
Implementing

working on loading

Commit:
Implementing

Initial work!

Fermain commented:
Did you get my Loom feedback? There is one conflict to be resolved still before we can merge this in

Othman37 committed:
Update README.MD

Ousman37 commented:
Can you please check if it is fine now?

Fermain commented:
This will go away when it's fine: 
<img width="977" alt="image" src="https://user-images.githubusercontent.com/10801540/213993410-349c41ca-92bf-492d-b816-2ec4dc727cb0.png">

Ousman37 committed:
Merge branch 'main' into workflow_Js2
